chainforge/react-server/src/store.js

```diff
@@ -238,8 +239,19 @@ const useStore = create((set, get) => ({
 
     // Pull data from each source recursively:
     const pulled_data = {};
+    let var_history = new Set();
     const get_outputs = (varnames, nodeId) => {
       varnames.forEach(varname => {
+        // Check for duplicate variable names
+        if (var_history.has(String(varname).toLowerCase())) {
+          throw new ProjectError({
```

Comment:
I would make the constructor of `DuplicateVariableNameError` take a variable name, and everything else (the message) should be set inside its constructor.